src/plugins/crypto/gpg.c

```diff
@@ -784,25 +784,37 @@ int CRYPTO_PLUGIN_FUNCTION (gpgCall) (KeySet * conf, Key * errorKey, Key * msgKe
 		if (msgKey)
 		{
 			close (STDIN_FILENO);
-			dup (pipe_stdin[0]);
+			if (dup (pipe_stdin[0]) < 0)
+			{
+				ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to redirect stdin.");
+				return -2;
```

Comment:
We have already forked here, this might not do what you expect. (It does not set an error in our process, same problem also 3x below.) See also #1590

Reply:
would it be sufficient to call `exit(-2)` like we do in `external.cpp:196`?